lib/ddtrace/configuration/components.rb

```diff
@@ -96,11 +109,62 @@ def build_tracer_options(settings)
             opts[:writer_options] = settings.writer_options if settings.writer.nil?
           end
         end
+
+        def build_profiler_recorder(settings)
+          event_classes = []
+
+          if settings.profiling.enabled && settings.profiling.cpu.enabled
+            event_classes << Datadog::Profiling::Events::StackSample
+          end
+
+          Datadog::Profiling::Recorder.new(event_classes, settings.profiling.max_events)
+        end
+
+        def build_profiler_collectors(settings, recorder)
+          return [] unless settings.profiling.enabled && settings.profiling.cpu.enabled
+
+          [
+            Datadog::Profiling::Collectors::Stack.new(
+              recorder,
+              enabled: settings.profiling.cpu.enabled
+              # TODO: Provide proc that identifies Datadog worker threads?
```

Comment:
It's a good point, but the Datadog worker could be a bottleneck and that's something we'd be very interested in knowing 🙂